app/(dashboard)/enroll/[repositoryId]/actions.ts

```diff
@@ -61,3 +62,23 @@ export const hasEnrollmentForRepositoryAction = async (repositoryId: string): Pr
     throw new Error("Failed to check enrollment.");
   }
 };
+
+/**
+ * Retrieves repository data for a given repository ID.
+ * @param repoId - The ID of the repository to retrieve.
+ * @returns The repository object.
+ */
+export const getRepositoryDataAction = async (repoId: string) => {
```

Comment:
even though the server action is located in the dashboard route it has no authentication or authorization. A server action is essentially just a public endpoint. Even though the `Repository` doesn't seem to have sensitive information it would make sense to add authentication and authorization to every server action if possible.